projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -669,7 +669,15 @@ def __verify_payload_format(payload_dict: dict):
         # Check if payload dict is valid json
         # TODO: optimize the check - we should not need to serialize the payload every time
         try:
-            json.dumps(payload_dict)
+            from decimal import Decimal
+
+            class DecimalJsonEncoder(json.JSONEncoder):
+                def default(self, obj):
+                    if isinstance(obj, Decimal):
+                        return float(obj)
```

Comment:
Why does this work? Should we not get rounding errors ?

Reply:
What rounding errors would we get?

Reply:
It depends on behaviour of float. 
Decimal is a  fixed point representation of a double (real number). While float is a floating point representation of a number. 
Sometimes when you cast integer or decimal to float it may be different e.g: cast(1.5 as float) == 1.500000000000001 
It depends - see this problem in c# https://stackoverflow.com/questions/8284275/casting-float-to-decimal-loses-precision-in-c-sharp 

So my question for you is it possible to have 

```
string_json = json.dumps(dict_with_decimal, DecimalJsonEncoder)
json_back = json.laods(string_json). 
```
Could some of the above situation happens? If yes, we need to document it in DecimalJsonEncoder

Reply:
I'll include a comment; I don't think this issue would be a problem as we only use this class for the check, not for any actual serialization.